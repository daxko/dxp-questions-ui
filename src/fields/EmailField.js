var React = require('react');

var EmailField = React.createClass({

	onKeyPress: function(e) {
		if (e.key == "<" || e.key == ">")
			e.preventDefault();
	},

	render: function() {
		return (
			<input type="text" 
				maxLength={100} 
				readOnly={this.props.question.read_only}
				value={this.props.answer || ''}
				onChange={this.props.onChange}
				onKeyPress={this.onKeyPress}
				className={this.props.className}
				onBlur={this.props.onBlur}
			/>
		);
	}

});

module.exports = EmailField;