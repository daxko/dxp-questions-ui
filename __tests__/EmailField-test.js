import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import EmailField from '../src/fields/EmailField'
const json = require('../snippets/json/email_question.json')

configure({adapter: new Adapter() })

describe('EmailField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<EmailField question={json}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
