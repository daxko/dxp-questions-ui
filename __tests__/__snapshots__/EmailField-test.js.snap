// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailField
    question={
      Object {
        "description": "We will not send unsolicited emails.",
        "max_length": 100,
        "pattern": "^\\\\w+([-+.']\\\\w+)*@\\\\w+([-.]\\\\w+)*\\\\.\\\\w+([-.]\\\\w+)*$",
        "read_only": false,
        "required": true,
        "title": "E-mail address",
        "type": "email",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "maxLength": 100,
      "onBlur": undefined,
      "onChange": undefined,
      "onKeyPress": [Function],
      "readOnly": false,
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "maxLength": 100,
        "onBlur": undefined,
        "onChange": undefined,
        "onKeyPress": [Function],
        "readOnly": false,
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
